### DISCUSSION TITLE
Trigger LDAP-Sync via REST-API

### DISCUSSION THREAD

**Discussion by noraab** [keycloak/keycloak#21977] [Q&A]
As I couldn't find anything in the docs: Is there any REST-endpoint available to trigger a sync with an LDAP User Federation Provider?
Thank you.

**Comment by jhollmannk** [comment]
We do that in our application. We use the Keycloak admin-client for that in Java but the REST endpoint should be

admin/realms/{realm}/user-storage/{ldap-provider-uuid}/sync?action=triggerFullSync

Getting the UUID of the LDAP storage provider is a bit complicated. We get all components via

admin/realms/{realm}/components

and look for an entry with "providerType": "org.keycloak.storage.UserStorageProvider".

Hope that helps.
